lib/matplotlib/backends/backend_pgf.py

```diff
@@ -987,4 +1007,215 @@ def _cleanup_all():
+        self._file = open(self._fname_tex, 'wb')
+
+    def _write_header(self, width_inches, height_inches):
+        supported_keys = {
+            'title', 'author', 'subject', 'keywords', 'creator',
+            'producer', 'trapped'
+        }
+        infoDict = {
+            'creator': 'matplotlib %s, http://matplotlib.org' % __version__,
+            'producer': 'matplotlib pgf backend %s' % __version__,
+        }
+        metadata = {k.lower(): v for k, v in self.metadata.items()}
+        infoDict.update(metadata)
+        hyperref_options = ''
+        for k, v in infoDict.items():
+            if k not in supported_keys:
+                raise ValueError(
+                    'Not a supported pdf metadata field: "{}"'.format(k)
+                )
+            hyperref_options += 'pdf' + k + '={' + str(v) + '},'
+
+        latex_preamble = get_preamble()
+        latex_fontspec = get_fontspec()
+        latex_header = r"""\PassOptionsToPackage{{
+  {metadata}
+}}{{hyperref}}
+\RequirePackage{{hyperref}}
+\documentclass[12pt]{{minimal}}
+\usepackage[
+    paperwidth={width}in,
+    paperheight={height}in,
+    margin=0in
+]{{geometry}}
+{preamble}
+{fontspec}
+\usepackage{{pgf}}
+\setlength{{\parindent}}{{0pt}}
+
+\begin{{document}}%%
+""".format(
+            width=width_inches,
+            height=height_inches,
+            preamble=latex_preamble,
+            fontspec=latex_fontspec,
+            metadata=hyperref_options,
+        )
+        self._file.write(latex_header.encode('utf-8'))
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        """
+        Finalize this object, running LaTeX in a temporary directory
+        and moving the final pdf file to `filename`.
+        """
+        self._file.write(rb'\end{document}\n')
+        self._file.close()
+
+        if self._n_figures > 0:
+            try:
+                self._run_latex()
+            finally:
+                try:
+                    shutil.rmtree(self._tmpdir)
+                except:
+                    TmpDirCleaner.add(self._tmpdir)
+        elif self.keep_empty:
+            open(self._outputfile, 'wb').close()
+
+    def _run_latex(self):
+        texcommand = get_texcommand()
+        cmdargs = [
+            str(texcommand),
+            "-interaction=nonstopmode",
+            "-halt-on-error",
+            os.path.basename(self._fname_tex),
+        ]
+        try:
+            check_output(cmdargs, stderr=subprocess.STDOUT, cwd=self._tmpdir)
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError(
+                "%s was not able to process your file.\n\nFull log:\n%s"
+                % (texcommand, e.output.decode('utf-8')))
+
+        # copy file contents to target
+        with open(self._fname_pdf, "rb") as fh_src:
+            with open(self._outputfile, "wb") as fh:
```

Comment:
These could be combined into one:

`with open(...) as fh_src, open(...) as fh:`

But if that clashes with the pep8 line-length limit, I think I like nested `with`s better than escaping the newline.

Reply:
Yes, I had it on one line before, was 84 chars

Reply:
just use copyfile instead?

Reply:
@MaxNoe this comment